src/main/java/com/easypost/http/Requestor.java

```diff
@@ -430,14 +429,51 @@ private static EasyPostResponse makeURLConnectionRequest(final RequestMethod met
      * @throws GatewayTimeoutError     when the request fails due to a gateway timeout.
      * @throws UnknownApiError         when the request fails due to an unknown API error.
      */
-    public static <T> T request(final RequestMethod method, String url, final Map<String, Object> params,
+    public static <T> T request(final RequestMethod method, final String endpoint, final Map<String, Object> params,
                                 final Class<T> clazz, final EasyPostClient client)
+            throws GatewayTimeoutError, RateLimitError, InvalidRequestError, NotFoundError, TimeoutError, EncodingError,
+            UnauthorizedError, MethodNotAllowedError, InternalServerError, UnknownApiError, ServiceUnavailableError,
+            ForbiddenError, JsonError, HttpError, RedirectError, PaymentError {
+        String apiVersion = client.getApiVersion();
+        return request(method, endpoint, params, clazz, client, apiVersion);
+    }
+
+    /**
+     * Send an HTTP request to EasyPost.
+     *
+     * @param <T>        Any class.
+     * @param method     The method of the API request.
+     * @param endpoint   The endpoint of the API request.
+     * @param params     The params of the API request.
+     * @param clazz      The class of the object for deserialization
+     * @param client     The EasyPostClient object.
+     * @param apiVersion The API version to use for this request.
+     * @return A clazz-type object.
+     * @throws HttpError               when the HTTP connection cannot be made.
+     * @throws EncodingError           when the request query cannot be encoded.
+     * @throws JsonError               when the request body cannot be encoded.
+     * @throws RedirectError           when the request is redirected.
+     * @throws UnauthorizedError       when the request is unauthorized.
+     * @throws ForbiddenError          when the request is forbidden.
+     * @throws PaymentError            when the request requires payment.
+     * @throws NotFoundError           when the request endpoint is not found.
+     * @throws MethodNotAllowedError   when the request method is not allowed.
+     * @throws TimeoutError            when the request times out.
+     * @throws InvalidRequestError     when the request is invalid.
+     * @throws RateLimitError          when the request exceeds the rate limit.
+     * @throws InternalServerError     when the request fails due to an internal server error.
+     * @throws ServiceUnavailableError when the request fails due to a service unavailability.
+     * @throws GatewayTimeoutError     when the request fails due to a gateway timeout.
+     * @throws UnknownApiError         when the request fails due to an unknown API error.
+     */
+    public static <T> T request(final RequestMethod method, final String endpoint, final Map<String, Object> params,
```

Comment:
I am fine with this PR. The only concern I have is this public function because it takes in the `apiVersion` parameter, meaning that users can use this function with any `apiVersion`. I believe people will probably not use this `request` function anyway but just wanted to bring this up.

Reply:
Fair enough, I'll take a crack at making this function inaccessible to end-users since, you're right, no one should be manually calling this.

Reply:
I don't think this will be a problem, since, as you said, users shouldn't ever have a need to call this method. But if they did, they could technically hit a different endpoint, as well as a different API version. What I'm saying is, the additional API version parameter doesn't introduce any more risk or anti-pattern than was already present.

I would love for this to be all internal-only access, but as we've discussed before, this method has to be public since it is in a different package/namespace than the services. Such is Java.

Reply:
We should avoid this if we could. One way to solve this is by passing another client with `Beta` apiVersion when initializing `BetaReferralCustomerService` in the EasyPostClient class constructor. This way, we can remove this function.

Reply:
Yeah, we should do it that way, I believe that was part of the spec for our rewrite design on how to handle beta functionality, but I'm not seeing a `BetaClient` in the library (a la C#: https://github.com/EasyPost/easypost-csharp/blob/master/EasyPost/BetaClient.cs)

Thoughts @Justintime50 ?

Reply:
My suggestion would be to take the path of least resistance. In PHP for instance, we simply have a beta flag on the requestor function that tells it to hit the beta endpoint instead. I would recommend going a similar route.

Reply:
Good, so what we have here is good to go.